test/unicode/utf8.jl

```diff
@@ -10,3 +10,27 @@ let ch = 0x10000
         end
     end
 end
+
+let str = UTF8String(b"this is a test\xed\x80")
+    @test next(str, 15) == ('\ufffd', 16)
+    @test_throws BoundsError getindex(str, 0:3)
+    @test_throws BoundsError getindex(str, 17:18)
+    @test_throws BoundsError getindex(str, 2:17)
+    @test_throws UnicodeError getindex(str, 16:17)
+    @test string(Char(0x110000)) == "\ufffd"
+    sa = SubString{ASCIIString}("This is a silly test", 1, 14)
+    s8 = convert(SubString{UTF8String}, sa)
+    @test typeof(s8) == SubString{UTF8String}
+    @test s8 == "This is a sill"
+    @test convert(UTF8String, b"this is a test\xed\x80\x80") == "this is a test\ud000"
+end
+
+# Test undocumented, unexported function, used in a registered package
+@test Base.is_utf8_start(0x65)
```

Comment:
(and not a stupid question either!  I especially didn't want to break JuliaParse right yet, which depends on the utf8sizeof [which I don't think is even correct], because Coverage depends on it!)